### PR TITLE
:hammer: Collections Category/Group fixes

### LINKED ISSUE #169: Collection categories are broken
This is apparently a Jekyll bug, but collections use their sub-folder names as their `category` even if you specify one. That causes our https://2022.djangocon.us/speaking/twitter/, our youtube meta info, and other pages to show rooms and other info we don't need for tweeting about talks. 

The only solution I can think of is:
- [x] add another "name for category"
- [x] add the field to our process.py models
- [x] setup an `__init__` to fill in "name for category" if it's blank
- [x] re-run this script for our project
- [x] update speaking* templates to look for this new field instead of category
- [x] publish!

### PR DESCRIPTION
**Pull request by jefftriplett**
Closes #169. 

This is a workaround for Jekyll Collections (no clue when this changed) using category == subfolder instead of what we assign it. I updated `process.py` to validate/fix/sync and re-ran it over our schedule data.

### CODE REVIEW

File: _schedule/talks/2022-10-17-09-30-t0-opening-remarks.md [hunk 1]
```diff
@@ -3,8 +3,9 @@ accepted: true
 category: talks
 date: 2022-10-17 09:30:00-07:00
 end_date: 2022-10-17 09:45:00-07:00
+group: talks
 layout: session-details
-permalink: /talks/opening-remarks-monday/
+permalink: /talks/opening-remarks/
```
Comment (drewbrew) [inline _schedule/talks/2022-10-17-09-30-t0-opening-remarks.md]: We need to keep the separate slugs for each day of lightning talks. I'll make a separate PR to adjust that.

File: _schedule/talks/2022-10-18-09-45-t0-come-on-in-the-waters-fine-making-python.md [hunk 2]
```diff
@@ -4,12 +4,13 @@ category: talks
 date: 2022-10-18 09:45:00-07:00
 difficulty: All
 end_date: 2022-10-18 10:30:00-07:00
+group: talks
 layout: session-details
 link: null
-permalink: /talks/keynote-come-on-in-waters-fine/
+permalink: /talks/come-on-in-the-waters-fine-making-python/
```
Comment (drewbrew) [inline _schedule/talks/2022-10-18-09-45-t0-come-on-in-the-waters-fine-making-python.md]: We need to update the blog post that links to this talk

File: _schedule/talks/2022-10-18-18-00-board-game-night.md [hunk 3]
```diff
@@ -2,14 +2,15 @@
...
-category: social event
+category: social-event
 date: 2022-10-18 18:00:00-07:00
 difficulty: null
 end_date: 2022-10-18 22:00:00-07:00
+group: social-event
 layout: session-details
-schedule_layout: full
-permalink: /social-event/board-game-night/
+permalink: /social-events/board-game-night/
```
Comment (drewbrew) [inline _schedule/talks/2022-10-18-18-00-board-game-night.md:11]: I need to check if the board game night post links here

File: _schedule/talks/2022-10-19-09-45-t0-keynote-state-of-the-object-relational.md [hunk 4]
```diff
@@ -4,9 +4,10 @@ category: talks
 date: 2022-10-19 09:45:00-07:00
 difficulty: All
 end_date: 2022-10-19 10:30:00-07:00
+group: talks
 layout: session-details
 link: null
-permalink: /talks/keynote-state-of-orm
+permalink: /talks/keynote-state-of-the-object-relational/
```
Comment (drewbrew) [inline _schedule/talks/2022-10-19-09-45-t0-keynote-state-of-the-object-relational.md]: Need to check the blog here as well

**Review by drewbrew — APPROVED** [review]
Couple of slug changes that need human intervention. Looks good otherwise

**Comment by jefftriplett** [comment]
@drewbrew good catch. I'm won't merge until we sort it out. 

This has bitten us in the past, and I see a one-liner that I think will fix it going forward.  I'll push and fix.

File: _schedule/talks/2022-10-18-09-30-t0-opening-remarks.md [hunk 5]
```diff
@@ -3,8 +3,9 @@ accepted: true
 category: talks
 date: 2022-10-18 09:30:00-07:00
 end_date: 2022-10-18 09:45:00-07:00
+group: talks
 layout: session-details
-permalink: /talks/opening-remarks-tuesday/
+permalink: /talks/opening-remarks/
```
Comment (jefftriplett) [inline _schedule/talks/2022-10-18-09-30-t0-opening-remarks.md]: This is another we need to fix.

File: _schedule/talks/2022-10-17-09-30-t0-opening-remarks.md (hunk 1 above)
Comment (jefftriplett) [reply _schedule/talks/2022-10-17-09-30-t0-opening-remarks.md]: Fixed!

File: _schedule/talks/2022-10-18-09-45-t0-come-on-in-the-waters-fine-making-python.md (hunk 2 above)
Comment (jefftriplett) [reply _schedule/talks/2022-10-18-09-45-t0-come-on-in-the-waters-fine-making-python.md]: Fixed!

File: _schedule/talks/2022-10-19-09-45-t0-keynote-state-of-the-object-relational.md (hunk 4 above)
Comment (jefftriplett) [reply _schedule/talks/2022-10-19-09-45-t0-keynote-state-of-the-object-relational.md]: Reverted/fixed!

File: _schedule/talks/2022-10-18-18-00-board-game-night.md (hunk 3 above)
Comment (jefftriplett) [reply _schedule/talks/2022-10-18-18-00-board-game-night.md:11]: I didn't see it referenced in the post.

**Comment by jefftriplett** [comment]
Will follow up with a better perma-fix, but short term the process.py script won't regenerate permalinks once they are set.

File: _schedule/talks/2022-10-18-09-45-t0-come-on-in-the-waters-fine-making-python.md (hunk 2 above)
Comment (jefftriplett) [reply _schedule/talks/2022-10-18-09-45-t0-come-on-in-the-waters-fine-making-python.md]: Reverted back to the OG.

**Pull request merged**